draft-ietf-tls-esni.md

```diff
@@ -163,44 +163,35 @@ and backend servers are physically separated.
 
 ## Encrypted ClientHello (ECH)
 
-ECH allows the client to encrypt sensitive ClientHello extensions, e.g., SNI,
-ALPN, etc., under the public key of the client-facing server. This requires the
-client-facing server to publish the public key and metadata it uses for ECH for
-all the domains for which it serves directly or indirectly (via Split Mode).
-This document defines the format of the ECH encryption public key and metadata,
-referred to as an ECH configuration, and delegates DNS publication details to
-{{!HTTPS-RR=I-D.ietf-dnsop-svcb-https}}, though other delivery mechanisms are
-possible. In particular, if some of the clients of a private server are
-applications rather than Web browsers, those applications might have the public
-key and metadata preconfigured.
-
-When a client wants to establish a TLS session with the backend server, it
-constructs its ClientHello as indicated in {{real-ech}}. We will refer to
-this as the ClientHelloInner message. The client encrypts this message using
-the public key of the ECH configuration. It then constructs a new ClientHello,
-the ClientHelloOuter, with innocuous values for sensitive extensions, e.g., SNI,
-ALPN, etc., and with an "encrypted_client_hello" extension, which this document
-defines ({{encrypted-client-hello}}). The extension's payload carries the
-encrypted ClientHelloInner and specifies the ECH configuration used for
-encryption. Finally, it sends ClientHelloOuter to the server.
-
-Upon receiving the ClientHelloOuter, a TLS server takes one of the following
-actions:
-
-1. If it does not support ECH, it ignores the "encrypted_client_hello" extension
-   and proceeds with the handshake as usual, per {{RFC8446, Section 4.1.2}}.
-1. If it is a client-facing server for the ECH protocol, but cannot decrypt the
-   extension, then it terminates the handshake using the ClientHelloOuter. This
-   is referred to as "ECH rejection". When ECH is rejected, the client-facing
-   server sends an acceptable ECH configuration in its EncryptedExtensions
-   message.
-1. If it supports ECH and decrypts the extension, it forwards the
-   ClientHelloInner to the backend server, who terminates the connection. This
-   is referred to as "ECH acceptance".
+A client-facing server enables ECH by publishing an ECH configuration, which
+is an encryption public key and associated metadata. The server must publish
+this for all the domains it serves directly or indirectly (via Split Mode).
```

Comment:
```suggestion
this for all the domains it serves directly or indirectly via Shared or Split Mode, respectively.
```

Reply:
Oh, I got this from the earlier version of the text. I wonder whether both OR clauses are even necessary. WDYT about:

> for all the domains it serves via Shared or Split Mode.

Reply:
Applied that version and rebased.

Reply:
Lovely :-) thanks!